File: system/src/main/scala/com/anjunar/jaxrs/types/QueryTable.scala
package com.anjunar.jaxrs.types

import com.anjunar.scala.mapper.annotations.PropertyDescriptor
import java.util
import scala.annotation.meta.field
import scala.beans.BeanProperty

class QueryTable[S, E](@BeanProperty
                       @(PropertyDescriptor @field)(title = "Search")
                       var search : S,
                       rows: util.List[E],
                       size: Long) extends Table[E](rows, size) {

  def this() = this(null.asInstanceOf[S], null, 0L)

}

File: application/src/main/javascript/workspace/projects/shared/src/lib/domain/container/QueryTable.ts
import Entity from "../../mapper/annotations/Entity";
import Basic from "../../mapper/annotations/Basic";
import type LinkContainer from "./LinkContainer";
import ActiveObject from "./ActiveObject";
import Links from "./Links";
import OneToMany from "../../mapper/annotations/OneToMany";

@Entity("QueryTable")
export default class QueryTable<S, R> extends ActiveObject implements Links {

    override $type = "QueryTable"

    @OneToMany()
    rows : R[]

    @Basic()
    size : number

    @Basic()
    links : LinkContainer

    @Basic()
    search : S

}